docs/src/main/asciidoc/qute-reference.adoc

```diff
@@ -2559,6 +2559,64 @@ goodbye=Best regards, {name} <1>
 
 Message templates are validated during the build. If a missing message template is detected, an exception is thrown and build fails.
 
+=== Registration of Additional Sections
+
+An additional `SectionHelperFactory` can be registered manually via `EngineBuilder#addSectionHelper()` methods in a CDI observer method at runtime:
+
+[source,java]
+----
+import io.quarkus.qute.EngineBuilder;
+
+class MyBean {
+
+    void configureEngine(@Observes EngineBuilder builder) {
+       builder.addSectionHelper(new CustomSectionFactory());
+    }
+}
+----
+
+However, in this case the section factory is ignored during validation at build time.
+If you want to register a section that participates in validation of templates at build time then use the `@AdditionalSectionFactory` annotation:
```

Comment:
Shouldn’t it be named `AdditionalSectionHelperFactory` since it is meant to register new `SectionHelperFactory` objects?

Somehow I am not sure I like the `Additional` prefix, but that's my personal opinion :)

Reply:
I was thinking something in the line of JAX-RS's `@Provider` which can be reused for other objects too (but in this case meaning "register this CDI bean in the EngineBuilder")

Reply:
Well, we have `@Locate` for template locators but the meaning is a little bit different. In this case, it's not only about "register this CDI bean in the EngineBuilder". The main point is "validation of templates at build time".

Hm, the idea of generic annotation is interesting. On the other hand, a specific annotation makes it more clear (e.g. the javadoc can be very specific) and allows for furhter customization (such as `@Locate#value()` -> a regex matching all the template ids located).

Reply:
Maybe something like `@EngineConfiguration`?

Reply:
> Maybe something like `@EngineConfiguration`?

That sounds nicer actually, +1

Reply:
In theory, we could use it for `ValueResolver` and `NamespaceResolver` now and few others in the future (if needed). And we should keep `@Locate` because it's a bit specific.

Reply:
Bonus points if when used in `TemplateLocator`s the build fails asking to use `@Locate` :)

Reply:
How about now?